public/app/features/data-connections/tabs/DataSources/DataSourcesNew.tsx

```diff
@@ -0,0 +1,7 @@
+import React, { ReactElement } from 'react';
+
+import { NewDataSource } from 'app/features/datasources/components/NewDataSource';
+
+export function DataSourcesNew(): ReactElement | null {
```

Comment:
Does this component ever return `null`? 🤔

Reply:
Nah, you are right! 🙏 

Actually after our style-guide update these [don't need an explicit return type anymore](https://github.com/grafana/grafana/blob/main/contribute/style-guides/frontend.md#how-to-declare-functional-components).
I will update these 👍

Reply:
Fixed in https://github.com/grafana/grafana/pull/52436/commits/6ad0c8f37bd69672e04a5458bf43023dbb4b637e.